### DISCUSSION TITLE
[Help] - No visible @interface for 'FIRActionCodeInfo'

### DISCUSSION THREAD

**Discussion by CineMeApp** [invertase/react-native-firebase#4760] [Q&A] [answered]
Getting the following error in my react native ios app: 

`No visible @interface for 'FIRActionCodeInfo' declares the selector 'dataForKey:'
`

It appears to be in lines 798-805 of `RNFirebaseAuth.m`, or in this section:

```
if ([info dataForKey:FIRActionCodeEmailKey] != nil) {
        [data setValue:[info dataForKey:FIRActionCodeEmailKey] forKey:keyEmail];
      } else {
        [data setValue:[NSNull null] forKey:keyEmail];
      }

if ([info dataForKey:FIRActionCodeFromEmailKey] != nil) {
        [data setValue:[info dataForKey:FIRActionCodeFromEmailKey] forKey:@"fromEmail"];
      } else {
        [data setValue:[NSNull null] forKey:@"fromEmail"];
      }
```

I have the following podfile:
```
  - RNFBApp (10.4.0):
    - Firebase/CoreOnly (~> 7.3.0)
    - React-Core
  - RNFBAuth (10.4.1):
    - Firebase/Auth (~> 7.3.0)
    - React-Core
    - RNFBApp
  - RNFirebase (5.6.0):
    - Firebase/Core
    - React
    - RNFirebase/Crashlytics (= 5.6.0)
  - RNFirebase/Crashlytics (5.6.0):
    - Crashlytics
    - Fabric
    - Firebase/Core
    - React
```
with `react-native: 0.63.0` & `Xcode 12.3` on macOS Catalina

From the looks of things, a similar issue was addressed [here](https://github.com/invertase/react-native-firebase/issues/1304), but that was a few years ago now and my version looks up to date.

Any help is greatly appreciated!! 
Thank you

**Comment by andersonaddo** [comment, marked as answer]
Your podfile has some things that shouldn't be there. Check out https://github.com/invertase/react-native-firebase/blob/master/tests/ios/Podfile.lock to see what an up to date podfile should have

**Reply by CineMeApp** [reply]
Thank you! Always helps to see the actual podfile that has the correct versions *facepalm 
Thanks again!

**Comment by mikehardy** [comment]
You have mixed versions, do not under any circumstances do that. You should be on 10.4.1 of everything except App (which is at 10.4.0 because of #4708)

firebase-ios-sdk should be on 7.3.0 if you override it, but you probably don't need to override it

Once you are on the correct firebase-ios-sdk you will have the symbols you need.

Spending any time on old versions for troubleshooting is a poor investment. In future update all versions and verify prior to opening issues or discussions - much more efficient

**Reply by CineMeApp** [reply]
Awesome, thank you for your quick reply! Turns out I was using what I thought was the @react-native-firebase but it was something else instead, making development much harder than it needed to be.
Much appreciated!
